### PR TITLE
Add ubuntu-arm64 runner

### PR DESCRIPTION
**Pull request by aosoriodw**
# Description
When a webhook request contains tag `ubuntu-arm64`, provisioner will create a runner on a t4g.medium machine.
Runner will remove itself from the repository after the job finishes.
Runner will run for up to 6 hours. If it runs for longer than this, it will exit with an error and will not remove itself from the repository.

This PR also fixes the make targets used to test the provisioner locally

## Tests
Created runner using a local provisioner instance and verified that runner is able to execute a job and then exits. See [here](https://github.com/datawire/infra-actions/actions/runs/3389052795/jobs/5631721561) for an example.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,20 @@
 HOSTNAME := https://sw.bakerstreet.io
 DRY_RUN := true
+RUNNER_TAG := ''
+
+.PHONY: test-ubuntu-arm64
+test-ubuntu-arm64:
+	# SHA can be calculated like this:
+    #echo -n "payload=$(cat github-runner-provisioner/test/ubuntu-arm64_payload.json)" | openssl dgst -sha1 -hmac FAKE_TOKEN
```
Comment (AidanWenzel) [inline Makefile:8]: 👀 Thank you for adding this comment! Super helpful 🙌

File: github-runner-provisioner/runner.go [hunk 2]
```diff
@@ -5,6 +5,11 @@ import (
 	"github.com/aws/aws-sdk-go-v2/service/ec2"
 )
 
+var runners = map[string]func(context.Context, string, string, bool) error{
+	runnerLabel:            createMacM1Runner,
```
Comment (AidanWenzel) [inline github-runner-provisioner/runner.go:9]: should this be `macOSarm64` instead of `runnerLabel`?

**Review by AidanWenzel — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (aosoriodw) [reply Makefile:8]: My plan is to make the target smarter, so you can pass any token, but I'm not an expert and want to get this out ASAP.

File: github-runner-provisioner/runner.go (hunk 2 above)
Comment (aosoriodw) [reply github-runner-provisioner/runner.go]: Ohh, I forgot to refactor the MacM1 object. Good catch!

**Review by rhs — COMMENTED** [review]
LGTM, thanks for keeping the commits clean!

**Review by rhs — APPROVED** [review]

**Review by AidanWenzel — APPROVED** [review]

**Pull request merged**